src/components/bar-chart/bar-chart.tsx

```diff
@@ -0,0 +1,235 @@
+import Label from '../label';
+
+interface DataItem {
+	[key: string]: number | string;
+}
+interface Color {
+	fill: string;
+}
+
+interface BarChartProps {
+	/** An array of objects representing the source data for the chart. */
+	data: DataItem[];
+
+	/** An array of strings representing the keys to access data in each data object. Used for identifying different data series. */
+	dataKeys: string[];
+
+	/** An array of color strings that determine the colors for each data series in the chart. */
+	colors?: Color[];
+
+	/** Defines the layout of Bar Chart. */
+	layout?: 'horizontal' | 'vertical';
+
+	/** Defines are the chart bars are stacked. */
+	stacked?: boolean;
+
+	/** Whether to render the `<XAxis />` component for the x-axis. */
+	showXAxis?: boolean;
+
+	/** Whether to render the `<YAxis />` component for the y-axis. */
+	showYAxis?: boolean;
+
+	/** Toggle the visibility of the tooltip on hover, displaying detailed information for each data point. */
+	showTooltip?: boolean;
+
+	/** The tooltip indicator. It can be `dot`, `line` or `dashed`. */
+	tooltipIndicator?: 'dot' | 'line' | 'dashed';
+
+	/** Present tooltip lable key. E.g. for this data element: `{ month: 'January', desktop: 186, mobile: 80 }` if set lableKye to 'month' the tooltip will display the month name (like 'January'). */
+	tooltipLabelKey?: string;
+
+	/** Whether to render the `<Legend />` component to identify data series. */
+	showLegend?: boolean;
+
+	/** Whether to display the `<CartesianGrid />`, adding horizontal and vertical grid lines. */
+	showCartesianGrid?: boolean;
+
+	/** A function used to format the ticks on the axes, e.g., for formatting dates or numbers. */
+	tickFormatter?: ( value: string ) => string;
+
+	/** The key in the data objects representing values for the x-axis. This is used to access the x-axis values from each data entry. */
+	xAxisDataKey?: string;
+
+	/** The key in the data objects representing values for the y-axis. This is used to access the y-axis values from each data entry. */
+	yAxisDataKey?: string;
+
+	/** Font size for the labels on the x-axis. */
+	xAxisFontSize?: 'sm' | 'md' | 'lg';
+
+	/** Font color for the labels on the x-axis. */
+	xAxisFontColor?: string;
+
+	/** Font color for the labels on the y-axis. */
+	yAxisFontColor?: string;
+
+	/** Width of the chart container. */
+	chartWidth?: number;
+
+	/** Height of the chart container. */
+	chartHeight?: number;
+
+	/** Border radius of chart bar. */
+	borderRadius?: number;
+}
+
+const BarChart = ( {
+	data,
+	dataKeys = [],
+	colors = [],
+	layout = 'horizontal', // horizontal, vertical
+	stacked = false,
+	showXAxis = true,
+	showYAxis = true,
+	showTooltip = true,
+	tooltipIndicator = 'dot', // dot, line, dashed
+	tooltipLabelKey,
+	showLegend = false,
+	showCartesianGrid = true,
+	tickFormatter,
+	xAxisDataKey,
+	yAxisDataKey,
+	xAxisFontSize = 'sm', // sm, md, lg
+	xAxisFontColor = '#4B5563',
+	yAxisFontColor = '#4B5563',
+	chartWidth = 350,
+	chartHeight = 200,
```

Comment:
```suggestion
	width = 350,
	height = 200,
```

Reply:
@BabuTechBsf suggested to change prop name from width to something which is not generic. So I changed to ChartWidth, same for ChartHeight.